activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb

```diff
@@ -86,6 +86,18 @@ def supports_lazy_transactions?
         true
       end
 
+      def column_definitions(table_name)
```

Comment:
I am wondering, what would happen if MySQL considers a bug - https://github.com/rails/rails/issues/45832#issuecomment-1216443079, fixes that and then `0x` values would became valid values. 
But knowing MySQL, this can newer happen.

Reply:
Yeah, I agree, but my guess is we can add a `get_database_version < "8.0.fixed-version"` if that happens. I agree, I don't know how likely that is to happen.

Reply:
I think the risk to Rails users is probably pretty minimal. You'd have to have a default value of `0x`. I think it's probably sufficient that we test for that below (I really like having that test), and if MySQL changes their minds about how this is represented our CI can catch it.